### PR TITLE
[feat:filebeat/input/redis/slowlog] Add Redis replication role to slowlogs

### LINKED ISSUE #40486: [filebeat/input/redis/slowlog] Add field for the Redis replication role
**Describe the enhancement:**

Add an additional field to Redis slow logs (`role` / `redis_role`) which captures the replication role of Redis at the time of slow log gathering (either `master` or `slave`).

https://redis.io/docs/latest/commands/role/

**Describe a specific use case for the enhancement or feature:**

In our setup, we send computationally heavy read commands to replicas in a HA setup (backed by Redis Sentinel). When troubleshooting issues on the master, we want to be able to filter out slow log entries that were sent to replicas to reduce noise and to mitigate red herrings.

I would imagine an additional command `ROLE` would be sent [here](https://github.com/elastic/beats/blob/d7ae68c7aa40aa27deff6b90508bb8f8664a6efa/filebeat/input/redis/harvester.go#L86), then parsing back the result and adding it as a field.

### PR DESCRIPTION
**Pull request by jdheyburn**
- Enhancement

## Proposed commit message

See title

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] ~~I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Disruptive User Impact

No disruption expected

## Author's Checklist

- [ ] N/A

## Related issues

- Closes #40486

## Use cases

Filtering on slow logs created by the master, ignoring replicas.

### CODE REVIEW

File: filebeat/input/redis/harvester.go [hunk 1]
```diff
@@ -81,25 +81,32 @@ func (h *Harvester) Run() error {
 		return nil
 	default:
 	}
-	// Writes Slowlog get and slowlog reset both to the buffer so they are executed together
+	// Writes SLOWLOG GET, SLOWLOG RESET, and ROLE to the buffer so they are executed together
 	h.conn.Send("SLOWLOG", "GET")
```
Comment (mauri870) [inline filebeat/input/redis/harvester.go:85]: Looks like the linter is complaining about this code now, do you mind handling the errors returned by the Send calls? Thanks.

**Comment by ycombinator** [comment]
@jdheyburn In addition to @mauri870's feedback, there are also some merge conflicts now on this PR.  Would you mind addressing the feedback and resolving those conflicts by rebasing this PR on `main`?  Thanks!

**Comment by jdheyburn** [comment]
Sorry for the delay @mauri870 @ycombinator - hope the latest merge fixes those issues now 🙏🏻

File: filebeat/input/redis/harvester.go [hunk 2]
```diff
@@ -146,6 +155,7 @@ func (h *Harvester) Run() error {
 			"duration": mapstr.M{
 				"us": log.duration,
 			},
+			"role": role,
```
Comment (mauri870) [inline filebeat/input/redis/harvester.go:158]: Can you add an assertion [here](https://github.com/elastic/beats/blob/755831e/filebeat/tests/system/test_redis.py#L49) as well? Just so we get some coverage for this. Thanks.

`assert "redis.slowlog.role" in output`

Comment (jdheyburn) [reply filebeat/input/redis/harvester.go:158]: @mauri870 thanks! I've added that now, I don't see the test running in the checks though.

Comment (mauri870) [reply filebeat/input/redis/harvester.go:158]: I talked with the team and confirmed that it indeed runs in integration tests, but only after the PR is approved.

**Review by mauri870 — APPROVED** [review]

**Comment by mauri870** [comment]
buildkite test this

**Comment by jdheyburn** [comment]
It would be great to get this PR merged in ahead of the next filebeat release - thank you!

@mauri870 @rdner

**Comment by pierrehilbert** [comment]
run docs-build

**Review by rdner — APPROVED** [review]

**Comment by jdheyburn** [comment]
Hey team, this has been merged for a month but I've not seen it be released on the latest version: https://github.com/elastic/beats/releases/tag/v8.15.3

When can we expect it deployed out?

**Comment by ishleenk17** [comment]
> Hey team, this has been merged for a month but I've not seen it be released on the latest version: https://github.com/elastic/beats/releases/tag/v8.15.3
> 
> When can we expect it deployed out?

This has been backported to 8.x
Hence, it will be available in 8.16. It would not be available in 8.15.x releases.

**Comment by jdheyburn** [comment]
@ishleenk17 Thanks for the update, when is 8.16 due to be released?

**Pull request merged**